pkg/secretsencrypt/controller.go

```diff
@@ -30,6 +29,8 @@ const (
 	secretsProgressEvent       string = "SecretsProgress"
 	secretsUpdateCompleteEvent string = "SecretsUpdateComplete"
 	secretsUpdateErrorEvent    string = "SecretsUpdateError"
+
+	secretListPageSize = 20
```

Comment:
Why such a small page size? the default is 500

Reply:
This is tuned base on the resource utilization graphs at https://github.com/k3s-io/k3s/issues/10450. That issue doesn't have screenshots of the secrets-encrypt stuff specifically, but you can see the result of listing all snapshots with the default page size - it takes a TON of memory for the apiserver to collect all the objects, deserialize them from the storage format, convert and serialize them back into a result response body, send them to the client, have the client deserialize them again, and store them in memory until the function is done processing them. When etcd and the apiserver all run in separate processes the memory is spread out a bit more, but here in K3s we get dinged for every step of that being in the same process. We can keep the peak memory utilization much lower if we limit the page size.